### PR TITLE
feat: Add configurable liveness probe timeout

### LINKED ISSUE #917: Default helm liveness probe timeout (1s) is too short
After running `helm install --repo https://helm.dask.org dask-kubernetes-operator dask-kubernetes-operator --create-namespace -n dask-operator`, my `dask-operator` pod is continually being restarted due to failing the liveness probe: it seems like in practice it takes between 1 and 2s to respond. Patching the deployment and overriding the timeout to 5s resolved the issue.

Maybe the probe timeout could be made configurable in the helm template?

### PR DESCRIPTION
**Pull request by bnaul**
Fixes #917

Plain helm install:
```
Liveness:      http-get http://:8080/healthz delay=0s timeout=1s period=10s #success=1 #failure=3
```
With `--set livenessProbe.timeoutSeconds=5`:
```
Liveness:      http-get http://:8080/healthz delay=0s timeout=5s period=10s #success=1 #failure=3
```

### CODE REVIEW

File: dask_kubernetes/operator/deployment/helm/dask-kubernetes-operator/values.yaml [hunk 1]
```diff
@@ -54,6 +54,9 @@ affinity: {}  # Affinity
 
 priorityClassName: null  # Priority class
 
+livenessProbe:
+  timeoutSeconds: 1
```
Comment (bnaul) [inline dask_kubernetes/operator/deployment/helm/dask-kubernetes-operator/values.yaml:58]: @jacobtomlinson I would probably say bump the default but no strong preference
```suggestion
  timeoutSeconds: 5
```

**Review by jacobtomlinson — APPROVED** [review]
This looks great! Happy to bump the default value.

File: dask_kubernetes/operator/deployment/helm/dask-kubernetes-operator/values.yaml (hunk 1 above)
Comment (jacobtomlinson) [inline dask_kubernetes/operator/deployment/helm/dask-kubernetes-operator/values.yaml:58]:
```suggestion
  timeoutSeconds: 5
```

**Comment by jacobtomlinson** [comment]
Hmm it won't let me commit the suggestion to change it to `5`. Could you update the PR?

**Comment by bnaul** [comment]
@jacobtomlinson done, plus updated the README

**Review by jacobtomlinson — APPROVED** [review]
Perfect thank you!

**Pull request merged**